clients/src/main/java/org/apache/kafka/common/requests/OffsetsForLeaderEpochRequest.java

```diff
@@ -143,7 +154,7 @@ public OffsetsForLeaderEpochRequest(Map<TopicPartition, PartitionData> epochsByP
 
     public OffsetsForLeaderEpochRequest(Struct struct, short version) {
         super(ApiKeys.OFFSET_FOR_LEADER_EPOCH, version);
-        replicaId = struct.getOrElse(REPLICA_ID, CONSUMER_REPLICA_ID);
+        replicaId = struct.getOrElse(REPLICA_ID, DEBUGGING_REPLICA_ID);
```

Comment:
This was a separate bug, I guess? Might be worth mentioning in the PR description.

Reply:
Yeah, we missed this. When the follower has to use an old version of the protocol without the replicaId field, we need a sentinel value which allows reading to the log end. I chose this one for consistency with other APIs (such as ListOffsets). I will mention this in the description.